cpp/src/parquet/encryption/crypto_factory.cc

```diff
@@ -48,8 +51,20 @@ std::shared_ptr<FileEncryptionProperties> CryptoFactory::GetFileEncryptionProper
 
   std::shared_ptr<FileKeyMaterialStore> key_material_store = NULL;
   if (!encryption_config.internal_key_material) {
-    // TODO: using external key material store with Hadoop file system
-    throw ParquetException("External key material store is not supported yet.");
+    if (parquet_file_path == nullptr) {
+      std::stringstream ss;
+      ss << "Output file path cannot be null"
```

Comment:
It would be good to clarify this a bit:
```suggestion
      ss << "Output file path cannot be null when using external key material"
```